ledger/simulation/tracer.go

```diff
@@ -178,7 +245,33 @@ func (tracer *evalTracer) saveEvalDelta(evalDelta transactions.EvalDelta, appIDT
 	applyDataOfCurrentTxn.EvalDelta.InnerTxns = inners
 }
 
+func (tracer *evalTracer) makeOpcodeTraceUnit(cx *logic.EvalContext) OpcodeTraceUnit {
+	return OpcodeTraceUnit{PC: uint64(cx.PC())}
+}
+
 func (tracer *evalTracer) BeforeOpcode(cx *logic.EvalContext) {
+	defer func() {
```

Comment:
I think this code would be better placed in `AfterOpcode`, since we want to eventually report stack changes as well, and those won't be available yet here.

Although if the PC's already advanced in `AfterOpcode`, that might be problematic.

Regardless, I don't love the use of defer here. I imagine you're doing it because we bail out early for logicsigs, but I don't see why you couldn't run this code before the logicsig branch.